code/modules/paperwork/ticketmachine.dm

```diff
@@ -5,234 +5,212 @@
-		return
-	if((user in ticket_holders) && !(obj_flags & EMAGGED))
-		to_chat(user, "You already have a ticket!")
-		return
-	playsound(src, 'sound/machines/terminal_insert_disc.ogg', 100, FALSE)
-	ticket_number ++
-	to_chat(user, "<span class='notice'>You take a ticket from [src], looks like you're ticket number #[ticket_number]...</span>")
+	ready = FALSE
+	playsound(src, 'sound/machines/terminal_insert_disc.ogg', 100, 0)
+	addtimer(CALLBACK(src, .proc/reset_cooldown), cooldown)//Small cooldown to prevent the clown from ripping out every ticket
+	currentNum ++
+	to_chat(user, "<span class='notice'>You take a ticket from [src], looks like you're customer #[currentNum]...</span>")
 	var/obj/item/ticket_machine_ticket/theirticket = new /obj/item/ticket_machine_ticket(get_turf(src))
-	theirticket.name = "Ticket #[ticket_number]"
-	theirticket.maptext = "<font color='#000000'>[ticket_number]</font>"
-	theirticket.saved_maptext = "<font color='#000000'>[ticket_number]</font>"
-	theirticket.ticket_number = ticket_number
+	theirticket.name = "Ticket #[currentNum]"
 	theirticket.source = src
-	theirticket.owner = user
+	theirticket.ticket_number = currentNum
+	theirticket.update_icon()
 	user.put_in_hands(theirticket)
-	ticket_holders += user
-	tickets += theirticket
+	if(tickets.len<currentNum)
+		tickets.len = currentNum //this grows the size of the list as needed.
+	tickets[currentNum] = theirticket
 	if(obj_flags & EMAGGED) //Emag the machine to destroy the HOP's life.
-		ready = FALSE
-		addtimer(CALLBACK(src, .proc/reset_cooldown), cooldown)//Small cooldown to prevent piles of flaming tickets
 		theirticket.fire_act()
 		user.dropItemToGround(theirticket)
 		user.adjust_fire_stacks(1)
 		user.IgniteMob()
 		return
 
+/obj/machinery/ticket_machine/attackby(obj/item/O, mob/user, params)
+	if(user.a_intent == INTENT_HARM) //so we can hit the machine
+		return ..()
+
+	if(default_deconstruction_screwdriver(user, "ticketmachine_panel", "ticketmachine", O))
+		updateUsrDialog()
+		update_icon()
+		return TRUE
+
+	if(default_deconstruction_crowbar(O) || stat)
+		return TRUE
+
+	if(istype(O, /obj/item/ticket_machine_remote))
+		if (!linked)
+			var/obj/item/ticket_machine_remote/Z = O //typecasting!!
+			if (!Z.connection)
+				Z.connection=src
+				to_chat(user,"<span class='info'>You link the remote to the machine.</span>")
+				linked = TRUE
+				return TRUE
+			else
+				to_chat(user,"<span class='warning'>The remote is already linked to a ticket machine!</span>")
+
+		else
+			to_chat(user,"<span class='warning'>The ticket machine is already linked to a remote!</span>")
+
+	if(istype(O, /obj/item/ticket_machine_ticket))
+		to_chat(user, "<span class='warning'>You put [O] into the ticket machine's recycling bin.</span>")
+		qdel(O) //KMC put a delay here. I'm not so forgiving. You accidently shred your ticket, you cry. (Also the delay felt really clunky)
+		return TRUE
+
+/obj/machinery/ticket_machine/Destroy()
+	return ..()
```

Comment:
```suggestion
```
Why define something if all you are doing is calling the parent?

Reply:
Kar told me to do something in here and I forgot what